From 01f6fce9026608bdf808725d4fb41448b9766ad9 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:04 -0700
Subject: [PATCH 1628/1851] v4l: xilinx: sdirxss: Fix warnings for
 CONSTANT_EXPRESSION_RESULT

commit 9a65b7def79f3dc3cca96d67e727e3ecc250fc4c from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes the warnings of CONSTANT_EXPRESSION_RESULT.

The actual warnings are:

- xsdirxss_parse_of():
Event impossible_and: 	The "and" condition "format->vf_code != 0U &&
format->vf_code != 3U && format->vf_code != 1U && format->vf_code != 2U
&& (core->bpc == 10U && format->width != 10U) && (core->bpc == 12U &&
format->width != 12U)" can never be true because "core->bpc" cannot be
equal to two different values at the same time.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index 85a1df083078..deb53d2e58a6 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -2323,8 +2323,8 @@ static int xsdirxss_parse_of(struct xsdirxss_state *xsdirxss)
 		    format->vf_code != XVIP_VF_YUV_420 &&
 		    format->vf_code != XVIP_VF_YUV_444 &&
 		    format->vf_code != XVIP_VF_RBG &&
-		    (core->bpc == 10 && format->width != 10) &&
-		    (core->bpc == 12 && format->width != 12)) {
+		    ((core->bpc == 10 && format->width != 10) ||
+		     (core->bpc == 12 && format->width != 12))) {
 			dev_err(core->dev,
 				"Incorrect UG934 video format set.\n");
 			return -EINVAL;
-- 
2.31.1

